app/src/main/java/waiter/ClientConnection.java

```diff
@@ -0,0 +1,27 @@
+package waiter;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.net.Socket;
+
+public class ClientConnection implements Connection {
```

Comment:
Do we have a test for this?

Reply:
As a result of ClientConnection being a thin wrapper around minimal native java functionality and since I don't want to mock what I don't own, I didn't test this wrapper.

Reply:
There's actually quite a bit of important logic in here and there's no way for us to know if the server still works when we make changes to this class unless we manually test it which isn't ideal.

Reply:
This is a good point and a good way to go about testing this, I will look into injecting a reader and writer class and a reader and writer mock in tests

Reply:
Going to defer this to be included in a separate PR (the first/next PR), Potential ticket name (for tracking purposes) -> add-client-connection-test-coverage ticket, thank you all for the help!